### ISSUE TITLE
Unable to mappe objects and list that are null

### ISSUE THREAD

**Issue by agnerbo** [AutoMapper/AutoMapper#1229]
[Program.cs.txt](https://github.com/AutoMapper/AutoMapper/files/214527/Program.cs.txt)
When mapping a Generic list or an object that are null I get an exception.

```
AutoMapper.AutoMapperMappingException: Missing type map configuration or unsupported mapping.

Mapping types:
Object -> OtherClass3[]
System.Object -> ConsoleApplication5.OtherClass3[]

Destination path:
OtherClass1.TestOtherClass2.TestOtherClass2.ArrayOfTestOtherClass3.ArrayOfTestOtherClass3

Source value:
(null)
   at AutoMapper.MappingEngine.Map(ResolutionContext context)
   at AutoMapper.Mappers.TypeMapObjectMapperRegistry.PropertyMapMappingStrategy.MapPropertyValue(ResolutionContext context, Object mappedObject, PropertyMap propertyMap)
   at AutoMapper.Mappers.TypeMapObjectMapperRegistry.PropertyMapMappingStrategy.Map(ResolutionContext context)
   at AutoMapper.MappingEngine.Map(ResolutionContext context)
   at AutoMapper.Mappers.TypeMapObjectMapperRegistry.PropertyMapMappingStrategy.MapPropertyValue(ResolutionContext context, Object mappedObject, PropertyMap propertyMap)
   at AutoMapper.Mappers.TypeMapObjectMapperRegistry.PropertyMapMappingStrategy.Map(ResolutionContext context)
   at AutoMapper.MappingEngine.Map(ResolutionContext context)
   at AutoMapper.Mapper.MapCore(Object source, Type sourceType, Type destinationType, MappingOperationOptions options)
   at AutoMapper.Mapper.AutoMapper.IMapper.Map(Object source, Type sourceType, Type destinationType, Action`1 opts)
   at AutoMapper.Mapper.AutoMapper.IMapper.Map[TSource,TDestination](TSource source)
   at ConsoleApplication5.Program.Main(String[] args) in c:\Users\am\Documents\Visual Studio 2013\Projects\ConsoleApplication5\ConsoleApplication5\Program.cs:line 78
```

I've added a testprogram that illustrate the error. The error is in both version 4.2.0 and 4.2.1. The error wasn't in version 4.1.1. I've tried with both Static mapping (Mapper.CreateMap) and MappingConfiguration which is used in the example.

[Program.cs.txt](https://github.com/AutoMapper/AutoMapper/files/214528/Program.cs.txt)

**Comment by lbargaoanu** [comment]
This works with the MyGet build. But you're calling CreateMapper all over the place. That's wrong. I suppose you did it to test things.

**Comment by agnerbo** [comment]
The CreateMapper "all over" were because i have the static mapping implemented with an older version of Automapper 4.1.1. Then I updated automapper to 4.2.1 (nugetpackage) and started having problems. I've then tried to make a simple test program to illustrate the error. Then I tried to change it to the dynamic version with MappingConfiguration to see if there were any deferrence. 

That's why the test is a bit off.

Thank you for testing with the MyGet build. Didn't know there were an dev-build. 

I'll try the MyGet build myself and if that works i'll wait to update my production application till next Nuget version.
